core/src/main/java/org/elasticsearch/search/SearchService.java

```diff
@@ -494,11 +496,12 @@ final SearchContext createContext(ShardSearchRequest request, @Nullable Engine.S
         return context;
     }
 
-    public DefaultSearchContext createSearchContext(ShardSearchRequest request, TimeValue timeout, @Nullable Engine.Searcher searcher)
+    public DefaultSearchContext createSearchContext(ShardSearchRequest request, OriginalIndices originalIndices,
+                                                    TimeValue timeout, @Nullable Engine.Searcher searcher)
         throws IOException {
         IndexService indexService = indicesService.indexServiceSafe(request.shardId().getIndex());
         IndexShard indexShard = indexService.getShard(request.shardId().getId());
-        SearchShardTarget shardTarget = new SearchShardTarget(clusterService.localNode().getId(), indexShard.shardId());
+        SearchShardTarget shardTarget = new SearchShardTarget(clusterService.localNode().getId(), indexShard.shardId(), originalIndices);
```

Comment:
I don't think we need this here at all lets pass always null here to the `SearchShardTarget`